diff -rNu a/lora_pkt_fwd/Makefile b/lora_pkt_fwd/Makefile
--- a/lora_pkt_fwd/Makefile	2017-04-05 11:42:35.000000000 +0000
+++ b/lora_pkt_fwd/Makefile	2019-06-19 11:24:47.000000000 +0000
@@ -22,7 +22,7 @@
 CC := $(CROSS_COMPILE)gcc
 AR := $(CROSS_COMPILE)ar
 
-CFLAGS := -O2 -Wall -Wextra -std=c99 -Iinc -I.
+CFLAGS := -libpthread -O2 -Wall -Wextra -std=c99 -Iinc -I.
 VFLAG := -D VERSION_STRING="\"$(RELEASE_VERSION)\""
 
 ### Constants for Lora concentrator HAL library
diff -rNu a/lora_pkt_fwd/src/jitqueue.c b/lora_pkt_fwd/src/jitqueue.c
--- a/lora_pkt_fwd/src/jitqueue.c	2017-04-05 11:42:35.000000000 +0000
+++ b/lora_pkt_fwd/src/jitqueue.c	2019-06-19 11:28:57.000000000 +0000
@@ -29,7 +29,7 @@
 
 /* -------------------------------------------------------------------------- */
 /* --- PRIVATE MACROS ------------------------------------------------------- */
-
+#define _GNU_SOURCE     /* needed for qsort_r to be defined */
 /* -------------------------------------------------------------------------- */
 /* --- PRIVATE CONSTANTS & TYPES -------------------------------------------- */
 #define TX_START_DELAY          1500    /* microseconds */
